feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -280,19 +280,50 @@ func setupP4RTClient(ctx context.Context, args *testArgs) error {
 	return nil
 }
 
-// Function to compare and check if the expected table is present in RPC ReadResponse
-func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+func tableMapFromEntry(t *testing.T, table *p4_v1.Entity) (map[string]any, error) {
 
+	var toMap map[string]any
+	switch entry := table.Entity.(type) {
+	case *p4_v1.Entity_TableEntry:
+		ent, err := json.Marshal(entry)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert table entry to json: %w", err)
+		}
+
+		err = json.Unmarshal([]byte(string(ent)), &toMap)
+		if err != nil {
+			return nil, fmt.Errorf("unable to unmarshal table entry to map: %w", err)
+		}
+
+	default:
+		t.Logf("Not a table entry: %v", entry)
+	}
+	return toMap["TableEntry"].(map[string]any), nil
+}
+
+// Function to compare and check if the expected table is present in RPC ReadResponse.
+func verifyReadReceiveMatch(t *testing.T, expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
 	matches := 0
 	for _, table := range received_entry.Entities {
-		if diff := cmp.Diff(expected_update[0].Entity.Entity, table.Entity, protocmp.Transform()); diff != "" {
-			glog.Errorf("Table entry diff (-want +got): \n%s", diff)
-			continue
+		var tableMap map[string]any
+		tableMap, err := tableMapFromEntry(t, table)
+		if err != nil {
+			return fmt.Errorf("unable to convert table entry to json: %w", err)
+		}
+		delete(tableMap, "meter_config")
+		delete(tableMap, "counter_data")
+		var tableMap1 map[string]any
+		tableMap1, err = tableMapFromEntry(t, expected_update[0].Entity)
+		if err != nil {
+			return fmt.Errorf("Unable to convert table entry to json: %w", err)
+		}
+		if cmp.Equal(tableMap, tableMap1) {
```

Comment:
Oh, the entire json marshal/unmarshal is just so that you can check equality without spurious errors? Ok, so you don't need to do any of that. Just use cmp.Equal with options to customize the comparison. Should be something like:

```
if cmp.Equal(table, expected_update[0].Entity, protocmp.Transform(), protocmp.IgnoreFields(p4_v1.TableEntry{}, "meter_config", "counter_data")) {
```

Reply:
The only change required on the original code is this.

Reply:
Hi @greg-dennis we have tried this earlier and it has failed due to incompatibility issues with protocmp. 
Here is the function that I wrote:
`// Function to compare and check if the expected table is present in RPC ReadResponse.
func verifyReadReceiveMatch(t *testing.T, expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
	matches := 0
	for _, table := range received_entry.Entities {

		if cmp.Equal(table, expected_update[0].Entity, protocmp.Transform(), protocmp.IgnoreFields(p4_v1.TableEntry{}, "meter_config", "counter_data")) {
			t.Logf("Table match succesful")
			matches += 1
		}

	}
	if matches == 0 {
		return errors.New("No matches found")
	}
	return nil
}
`

This is the error I am seeing:
./base_p4rt_test.go:288:94: cannot use p4_v1.TableEntry{} (value of type "github.com/p4lang/p4runtime/go/p4/v1".TableEntry) as type protoreflect.ProtoMessage in argument to protocmp.IgnoreFields:
	"github.com/p4lang/p4runtime/go/p4/v1".TableEntry does not implement protoreflect.ProtoMessage (ProtoReflect method has pointer receiver)
FAIL	github.com/openconfig/featureprofiles/feature/experimental/p4rt/ate_tests/base_p4rt [build failed]
FAIL

Reply:
I also tried adding a switch case and passing that to the protocmp.IgnoreFields but it is returning the same error.

Reply:
Can you try &p4_v1.TableEntry{} instead of p4_v1.TableEntry{}?

Reply:
That worked! Thank you so much.